### PR TITLE
RF: Decomplexify configure_origins()

### PR DESCRIPTION
**Pull request by mih**
Previous code looked at two datasets simultaneously per iteration.
This could result in making a fetch attempt to a non-local repository,
because it would no check whether the second dataset was actually local.

The new code focuses on a single dataset at a time, and all checks are
performed on each one.

### CODE REVIEW

**Comment by kyleam** [comment]
> Previous code looked at two datasets simultaneously per iteration.
> This could result in making a fetch attempt to a non-local repository,
> because it would no check whether the second dataset was actually local.

Oy, good catch.

Rewrite looks good to me.

File: datalad/core/distributed/clone.py [hunk 1]
```diff
@@ -727,38 +727,58 @@ def configure_origins(cfgds, probeds, label=None):
...
+        return
+
+    # no need to reconfigure original/direct origin again
+    if cfgds != probeds:
+        # prevent duplicates
+        known_remote_urls = set(
+            cfgds.config.get(r + '.url', None)
+            for r in cfgds.config.sections()
+            if r.startswith('remote.')
+        )
```
Comment (kyleam) [inline datalad/core/distributed/clone.py]: Hmm, what's the scenario in which these duplicates would arise?  Is it when, in a chain of origins---say "cfgds -> origin A -> origin B -> origin C"---origin C's "origin" remote points back to B?

Comment (mih) [reply datalad/core/distributed/clone.py]: Yes, that is the scenario

**Comment by mih** [comment]
Thx @kyleam  for the review!

**Pull request merged**